src/main/java/org/broadinstitute/hellbender/tools/pon/coverage/pca/PCATangentNormalizationResult.java

```diff
@@ -0,0 +1,141 @@
+        writePreTangentNormalizationOutput(preTangentNormalizedFile, commandLine);
+        writeTangentBetaHats(tangentBetaHatsFile, commandLine);
+        writeTargetFactorNormalizedOutput(targetFactorNormalizedCountsFile, commandLine);
+    }
+
+    /**
+     * Writes the tangent-factor normalized outputs.
+     *
+     * Please note that this file is written in the target format originally used in recapseg.  In other words, it is
+     *  written as if it was target coverage.
+     */
+    private void writeTangentNormalizedOutput(final File file, final String commandLine) {
+        writeOutput(file, tangentNormalized, commandLine, "Tangent normalized coverage profile");
+    }
+
+    /**
+     * Writes the pre-tangent-normalization read counts if a file was provided for it.
+     *
+     * This file is written in the target format originally used in recapseg.  In other words, it is
+     *  written as if it was target coverage. If the output file is null, we do not write a file.
+     */
+    private void writePreTangentNormalizationOutput(final File file, final String commandLine) {
+        writeOutput(file, preTangentNormalized, commandLine, "Pre tangent normalized coverage profile");
+    }
+
+    /**
+     * Write the beta-hats if an output file was provided for it.
+     */
+    private void writeTangentBetaHats(final File file, final String commandLine) {
+        if (file == null) {
+            return;
+        }
+        final List<String> countColumnNames = targetFactorNormalizedCounts.columnNames();
+        final List<String> columnNames = new ArrayList<>(countColumnNames.size() + 1);
+        columnNames.add(PON_SAMPLE_BETA_HAT_COLUMN_NAME);
+        columnNames.addAll(countColumnNames);
+        final TableColumnCollection columns = new TableColumnCollection(columnNames);
+        try (final TableWriter<Integer> writer = TableUtils.writer(file, columns,
+                (i, dataLine) -> dataLine.append(Integer.toString(i)).append(tangentBetaHats.getRow(i)))) {
+            writer.writeComment("fileFormat  = tsv");
+            writer.writeComment("commandLine = " + commandLine);
+            writer.writeComment("title = Tangent normalization Beta Hats");
+            for (int i = 0; i < tangentBetaHats.getRowDimension(); i++) {
+                writer.writeRecord(i);
+            }
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotCreateOutputFile(file, ex.getMessage());
+        }
+    }
+
+    /**
+     * Writes the target-factor normalized outputs.
+     */
+    private void writeTargetFactorNormalizedOutput(final File file, final String commandLine) {
+        writeOutput(file, targetFactorNormalizedCounts, commandLine, "Target factor normalized target counts");
+    }
+
+    private void writeOutput(final File file, final ReadCountCollection counts, final String commandLine, final String title) {
+        if (file == null) {
```

Comment:
when would someone try to write to a `null` file?  Should this throw an exception instead?